mozart++/system.cxx/io.hpp

```diff
@@ -1,31 +1,45 @@
 /**
- * Mozart++ Template Library
+ * Mozart++ Template Library: System/IO
  * Licensed under MIT License
  * Copyright (c) 2020 Covariant Institute
  * Website: https://covariant.cn/
  * Github:  https://github.com/covariant-institute/
  */
 #pragma once
 
-#include <cstdio>
+#include <mozart++/core>
 #include <cstring>
 #include <cstdint>
-#include <mozart++/core/base.hpp>
+#include <cstdio>
 
 #ifdef MOZART_PLATFORM_WIN32
+
 #include <Windows.h>
 #include <io.h>
+
 #else
+
 #include <unistd.h>
+
 #endif
 
-// On MSVC, ssize_t is SSIZE_T
 #ifdef _MSC_VER
+
 #include <BaseTsd.h>
-using ssize_t = SSIZE_T;
+
 #endif
 
+namespace mpp_impl {
+    static constexpr int PIPE_READ = 0;
+    static constexpr int PIPE_WRITE = 1;
+}
+
 namespace mpp {
+#ifdef _MSC_VER
+    // On MSVC, ssize_t is SSIZE_T
+    using ssize_t = SSIZE_T;
+#endif
```

Comment:
Consider adding a `#else` to define `mpp::ssize_t`, and replace all occurance of `ssize_t` with `mpp::ssize_t`